### ISSUE TITLE
Add Ruby 1.9.3 Compilation Error

### ISSUE THREAD

**Issue by speck-drum** [remear/jewelrybox#170]
Trying to add Ruby 1.9.3 with JewelryBox results in the following compilation error (taken from the end of ~/.rvm/log/ruby-1.9.3-p385/make.log):

compiling regparse.c
regparse.c:582:15: error: implicit conversion loses integer precision: 'st_index_t' (aka 'unsigned long') to 'int' [-Werror,-Wshorten-64-to-32]
    return t->num_entries;
    ~~~~~~ ~~~^~~~~~~~~~~
1 error generated.
compiling regsyntax.c
make: **\* [regparse.o] Error 1

All my attempts (fiddling with Architectures and/or Linking; creating a new user account and trying to install Ruby 1.9.3 there; trying to build an earlier version of Ruby 1.9.3; and so on) failed. Any ideas?

I documented my setup on

http://chaos-zu-haus.de/JewelryBoxAddRubyFailure/

Thanks in advance,

Wolfgang H.

**Comment by remear** [comment]
First off, I'd like to thank you for filing such a detailed issue. I doubt I've ever seen someone supply so much information to make it easier to find a solution to the problem.

Do you have gcc-4.2 on your machine? I've seen this error related to attempting to compile Ruby with LLVM. You can try to use the new command supplied by rvm to automatically set up your environment with homebrew and the necessary packages for compiling Ruby properly. Just open up a terminal and run `rvm requirements run force` Essentially this does everything for you as listed in `rvm requirements`, the "Release Notes" area on the JewelryBox dashboard.

**Comment by speck-drum** [comment]
Thanks for your support. Instead of trying to install ruby 1.9.3 I installed the recently released ruby-2.0.0-p0. The installation (using JewelryBox) went smooth without any hickups.

FYI:

ruby-2.0.0-p0:

  ruby:
    interpreter:  "ruby"
    version:      "2.0.0p0"
    date:         "2013-02-24"
    platform:     "x86_64-darwin12.2.0"
    patchlevel:   "2013-02-24 revision 39474"
    full_version: "ruby 2.0.0p0 (2013-02-24 revision 39474) [x86_64-darwin12.2.0]"

  homes:
    gem:          "/Users/whusmann/.rvm/gems/ruby-2.0.0-p0"
    ruby:         "/Users/whusmann/.rvm/rubies/ruby-2.0.0-p0"

  binaries:
    ruby:         "/Users/whusmann/.rvm/rubies/ruby-2.0.0-p0/bin/ruby"
    irb:          "/Users/whusmann/.rvm/rubies/ruby-2.0.0-p0/bin/irb"
    gem:          "/Users/whusmann/.rvm/rubies/ruby-2.0.0-p0/bin/gem"
    rake:         "/Users/whusmann/.rvm/rubies/ruby-2.0.0-p0/bin/rake"

  environment:
    PATH:         "/Users/whusmann/.rvm/gems/ruby-2.0.0-p0/bin:/Users/whusmann/.rvm/gems/ruby-2.0.0-p0@global/bin:/Users/whusmann/.rvm/rubies/ruby-2.0.0-p0/bin:/Users/whusmann/.rvm/bin:/Users/whusmann/perl5/perlbrew/bin:/Users/whusmann/perl5/perlbrew/perls/perl-5.17.8/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin"
    GEM_HOME:     "/Users/whusmann/.rvm/gems/ruby-2.0.0-p0"
    GEM_PATH:     "/Users/whusmann/.rvm/gems/ruby-2.0.0-p0:/Users/whusmann/.rvm/gems/ruby-2.0.0-p0@global"
    MY_RUBY_HOME: "/Users/whusmann/.rvm/rubies/ruby-2.0.0-p0"
    IRBRC:        "/Users/whusmann/.rvm/rubies/ruby-2.0.0-p0/.irbrc"
    RUBYOPT:      ""
    gemset:       ""

Am 2013-02-16 um 02:34 schrieb Ben Mills notifications@github.com:
